src/wfs/src/test/java/org/geoserver/wfs/UpdateElementHandlerTest.java

```diff
@@ -0,0 +1,40 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfs;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.geotools.geometry.jts.WKTReader2;
+import org.junit.Test;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.MultiPoint;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+
+public class UpdateElementHandlerTest {
```

Comment:
Nice to have a unit test, but I'd feel more comfortable merging if there was also a end-to-end test on transaction. Pick any TransactionTest (there are 3, one per version) and adapt an existing test using the wrong type of geometry, check that the right service exception is returned.